editorconfig/fnmatch.py

```diff
@@ -216,5 +217,8 @@ def translate(pat, nested=False):
         else:
             is_escaped = False
     if not nested:
-        result += '\Z(?ms)'
+        if sys.version_info[0] < 3 or sys.version_info[1] < 6:
+            result += r'\Z(?ms)'
+        else:
+            result = r'(?s:%s)\Z' % result
```

Comment:
Can you explain why we have this line?

Reply:
I meant, is the if condition necessary?

Reply:
Looks like a generic expression is possible. I pushed my attempt.